### PR TITLE
fix: CA1829, CA1834, CA1846, CA1847

### PR DESCRIPTION
**Pull request by daniel-akili**
Related to #202

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs [hunk 1]
```diff
@@ -91,7 +91,10 @@ private SpdxComponent ConvertJObjectToSbomComponent(ProcessRequest processReques
             var name = document["name"]?.ToString();
             var spdxVersion = document["spdxVersion"]?.ToString();
 
-            if (rootElements != null && rootElements.Count() > 1)
+            if (rootElements == null || rootElements.Length <= 1)
+            {
+            }
+            else
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs:97]: The empty `if` looks a little odd. What about `if (rootElements?.Length > 1)`?

File: test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public void TestInitialize()
         [TestMethod]
         public async Task TestNlohmann()
         {
-            var spdxFile = @"{
+            var spdxFile = /*lang=json,strict*/ @"{
```
Comment (JamieMagee) [inline test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs:32]: What's this?

File: test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public async Task TestNlohmann()
         [TestMethod]
         public async Task TestTinyxmlAndResource()
         {
-            var spdxFile = @"{
+            var spdxFile = /*lang=json,strict*/ @"{
```
Comment (JamieMagee) [inline test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs:79]: Same here

**Review by JamieMagee — CHANGES_REQUESTED** [review]
Couple of small comments for CA1829. Other changes look good 👍

File: test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs (hunk 2 above)
Comment (daniel-akili) [reply test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs:32]: Will look for more documentation about the `JSON002` error that raised the issue

Comment (JamieMagee) [reply test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs:32]: We spoke about this offline and decided to mark `JSON002` as a suggestion until we can find some documentation about it.

Comment (JamieMagee) [inline test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs:32]: re-added?

**Review by JamieMagee — CHANGES_REQUESTED** [review]
Couple of comments on `JSON002` and also snapshot diffs.

File: test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs (hunk 2 above)
Comment (daniel-akili) [reply test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs]: was re-added from previous commit, but should be good now

File: src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs [hunk 4]
```diff
@@ -91,7 +91,7 @@ private SpdxComponent ConvertJObjectToSbomComponent(ProcessRequest processReques
             var name = document["name"]?.ToString();
             var spdxVersion = document["spdxVersion"]?.ToString();
 
-            if (rootElements.Length > 1)
+            if (rootElements != null && rootElements.Length > 1)
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs:94]: I'm not sure if this is valid syntax, but you might be able to combine this into a single expression.

```suggestion
            if (rootElements?.Length > 1)
```

File: src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs [hunk 5]
```diff
@@ -91,7 +91,7 @@ private SpdxComponent ConvertJObjectToSbomComponent(ProcessRequest processReques
             var name = document["name"]?.ToString();
             var spdxVersion = document["spdxVersion"]?.ToString();
 
-            if (rootElements != null && rootElements.Count() > 1)
+            if (rootElements?.Length > 1)
```
Comment (daniel-akili) [inline src/Microsoft.ComponentDetection.Detectors/spdx/Spdx22ComponentDetector.cs:94]: changing to this syntax builds successfully but does not pass the snapshot checks. Revert it?

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**